ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/tree/node/services/TreeNodeConverterImpl.java

```diff
@@ -47,17 +48,14 @@ public Model toModel(final TreeNode treeNode) {
 	}
 
 	private List<Statement> addTreeNodeStatements(TreeNode treeNode) {
-		List<Statement> statements = new ArrayList<>();
-		Resource currentFragmentId = createResource(treeNode.getFragmentId());
-
-		statements.add(createStatement(currentFragmentId, RDF_SYNTAX_TYPE, createResource(TREE_NODE_RESOURCE)));
-		treeNode.getRelations().forEach(treeRelation -> {
-			TreeRelationResponse treeRelationResponse = new TreeRelationResponse(treeRelation.treePath(),
-					ldesConfig.getBaseUrl() + treeRelation.treeNode(),
-					treeRelation.treeValue(), treeRelation.treeValueType(), treeRelation.relation());
-			statements.addAll(treeRelationResponse.convertToStatements(
-					treeNode.getFragmentId()));
-		});
+		List<TreeRelationResponse> treeRelationResponses = treeNode.getRelations().stream()
+				.map(treeRelation -> new TreeRelationResponse(treeRelation.treePath(),
+						ldesConfig.getBaseUrl() + treeRelation.treeNode(),
+						treeRelation.treeValue(), treeRelation.treeValueType(), treeRelation.relation()))
+				.toList();
+		TreeNodeInfoResponse treeNodeInfoResponse = new TreeNodeInfoResponse(treeNode.getFragmentId(),
+				treeRelationResponses);
+		List<Statement> statements = new ArrayList<>(treeNodeInfoResponse.convertToStatements());
```

Comment:
Do we need to clone this list again if 'convertToStatements' already creates a new list?

Reply:
Hi @Tomvbe,
Your remark made me think about something. One expects a set of statements from a treeNodeInfoResponse to be consistent and immutable, i.e. after a TreeNodeInfoResponse is converted to a set of statements it should be impossible to change or update the statements. So I added immutability, also to the other conversions.